docs/upgrade.md

```diff
@@ -0,0 +1,246 @@
+        - "--to-version=1.12.0"
+        # if required the image prefix of the pool deployments can be
+        # changed using the flag below, defaults to whatever was present on old
+        # deployments.
+        #- "--to-version-image-prefix=openebs/"
+        # if required the image tags for pool deployments can be changed
+        # to a custom image tag using the flag below, 
+        # defaults to the --to-version mentioned above.
+        #- "--to-version-image-tag=ci"
+
+        # VERIFY that you have provided the correct list of CSPC Names
+        - "cspc-stripe"
+
+        # Following are optional parameters
+        # Log Level
+        - "--v=4"
+        # DO NOT CHANGE BELOW PARAMETERS
+        env:
+        - name: OPENEBS_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        tty: true
+
+        # the image version should be same as the --to-version mentioned above
+        # in the args of the job
+        image: openebs/upgrade-amd64:1.12.0
+        imagePullPolicy: IfNotPresent
+      restartPolicy: OnFailure
+---
+```
+
+You can get the above yaml from [here](../examples/upgrade/cstor-cspc.yaml).
+
+To get the CSPC pool names you can use the command:
+```sh
+$ kubectl -n openebs get cspc
+NAME          HEALTHYINSTANCES   PROVISIONEDINSTANCES   DESIREDINSTANCES   AGE
+cspc-stripe   1                  1                      1                  3h22m
+```
+
+The status of the job can be verified by looking at the logs of the job pod. To get the job pod use the command:
+```sh
+$ kubectl -n openebs get pods -l job-name=cstor-cspc-upgrade
+NAME                               READY   STATUS   RESTARTS   AGE
+cstor-cspc-upgrade-2x4bv     1/1     Running    0          34s
+```
+ 
+```sh
+$ kubectl -n openebs logs -f cstor-cspc-upgrade-2x4bv
+I0714 12:37:09.747331       1 cstor_cspc.go:65] Upgrading cspc-stripe to 1.12.0-RC2
+I0714 12:37:10.062861       1 deployment.go:77] patching deployment cspc-stripe-k7cc
+I0714 12:40:11.493424       1 deployment.go:114] deployment cspc-stripe-k7cc patched successfully
+I0714 12:40:11.493476       1 cspi.go:73] patching cspi cspc-stripe-k7cc
+I0714 12:40:11.503801       1 cspi.go:93] cspi cspc-stripe-k7cc patched
+I0714 12:40:11.527764       1 cstor_cspi.go:285] Verifying the reconciliation of version for cspc-stripe-k7cc
+I0714 12:40:21.632513       1 cspc.go:75] patching cspc cspc-stripe
+I0714 12:40:21.682353       1 cspc.go:95] cspc cspc-stripe patched
+I0714 12:40:21.693266       1 cstor_cspc.go:190] Verifying the reconciliation of version for cspc-stripe
+I0714 12:40:31.701881       1 cstor_cspc.go:76] Successfully upgraded cspc-stripe to 1.12.0-RC2
+```
+
+## cStor CSI volumes
+
+These instructions will guide you through the process of upgrading cStor CSI volumes from `1.11.0` or later to a newer release up to `1.12.0`.
+
+### Prerequisites
+
+Before upgrading the volumes make sure the following prerequisites are taken care of:
+
```

Comment:
@shubham14bajpai scaledown of application is recommended or not?

Reply:
Added a note